FreeBSD support

mrbaseman opened the pull request:
I have started this work some time ago. At the current stage I have managed to configure, compile and run _openfortivpn_ on FreeBSD 11.
The last remaining issue is that FreeBSD doesn't have `pppd` anymore and uses a user-space tool called `ppp`. It doesn't take many options from the command line but rather from STDIN or from a config file. To my understanding of the [man page](https://www.freebsd.org/cgi/man.cgi?query=ppp&sektion=8) it should be possible to put all necessary parameters into `/etc/ppp/ppp.conf` and _openfortivpn_ should work with this. If this works, the only missing bit is to provide a working sample configuration and perhaps an update of the man page.
Many command line options of _openfortivpn_ concerning how `pppd` is called are currently not supported on BSD. At a later stage one can modify the way `ppp` is called and pipe setup commands into the client before sending it to the background. I just haven't figured out to do this correctly in combination with `forkpty`, but we can keep this as a separate task, once `ppp` has successfully opened a connection by feeding it with appropriate parameters from a config file.

mrbaseman commented:
oh, I have just seen that @mnencia was faster here: https://github.com/adrienverge/openfortivpn/pull/144

mnencia commented:
Your patch looks more elaborated than mine. I like the ability to indipendently handle the include and lib part.

DimitriPapadopoulos commented:
@mrbaseman So what do you think, should we merge those two patches? Use the one from **mnencia** which is simpler and perhaps good enough?

mrbaseman commented:
I would combine them: His approach is simpler and already containd the update of the docu and the Travis file, but I would like to include the autoconf test if an application linking against openssl can be compiled with the flags chosen. 
One could even check if the openssl version matches a supported range. A lot of things change with version 1.1.0 - I would have to check how far we are affected by this major version increment. 
Support for newer versions would definitely qualify for a new issue but a version check at configure time would fit well into the current one.

mnencia commented:
To ease the merge, I'll add this patch in my pull request.

mnencia commented:
I've added this to my pull request, but it fails to detect the multiarch library location.

mnencia commented:
It fails to detect libraries in linux because of multiarch https://github.com/adrienverge/openfortivpn/pull/144#issuecomment-308377833

mnencia commented:
I've fixed it. Please review.

DimitriPapadopoulos committed:
Use autoconf macros in #ifdef

DimitriPapadopoulos committed:
Use autoconf macros in #ifdef

Note that 'struct termios' is defined in <termios.h> (POSIX).
Some obscure systems might use <sys/termios.h> but we do not
support that for now.

DimitriPapadopoulos committed:
Merge branch 'master' into forkpty

mrbaseman committed:
avoid usage of '$^' in Makefile.am

it is not supported in the make shipped with Freebsd

mrbaseman committed:
in configure.ac use AM_SILENT_RULES only if it is available

mrbaseman committed:
header inclusion of sys/mutex.h and net/if_var.h  using autoconf variables

mrbaseman committed:
include mach/mach.h based on autoconf variable

mrbaseman committed:
for inclusion of termios.h and libutil.h use autoconf variables

mrbaseman committed:
supppress compiler warnings in clang specifically for clang

mrbaseman committed:
determine existence of /proc/net/route via configure script

mrbaseman committed:
determine existence of netstat for osx and bsd via configure script

mrbaseman committed:
use autoconf variable about existence of /proc/net/route

mrbaseman committed:
once more use __clang__ for clang specific pragma

mrbaseman committed:
from configure script check whether rtentry is available and has rt_dst

mrbaseman committed:
a few missing bits for FreeBSD support, e.g. ppp instead of pppd

mrbaseman commented:
sorry, @DimitriPapadopoulos I wanted to create the pull request in Adrien's repository, not in yours, but since you have already opened a pull request for a few of the changes that I had in my branch, I thought it would make sense to rebase my work on yours, rather than having separate commits for the same changes in my branch.
